File: Heat/Controllers/WarehousesController.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.Entity
Imports System.Linq
Imports System.Net
Imports System.Web
Imports System.Web.Mvc
Imports Heat.Models
Imports Heat.Repositories

Namespace Controllers
    Public Class WarehousesController
        Inherits System.Web.Mvc.Controller

        Private _db As HeatDBContext

        Public Sub New(context As IHeatDBContext)
            _db = context
        End Sub


        ' GET: Warehouses
        Function Index() As ActionResult
            Return View(_db.Warehouses.ToList())
        End Function

        ' GET: Warehouses/Details/5
        Function Details(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim warehouse As Warehouse = _db.Warehouses.Find(id)
            If IsNothing(warehouse) Then
                Return HttpNotFound()
            End If
            Return View(warehouse)
        End Function

        ' GET: Warehouses/Create
        Function Create() As ActionResult
            Return View()
        End Function

        ' POST: Warehouses/Create
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Create(<Bind(Include:="ID,Code,Description,HasValue")> ByVal warehouse As Warehouse) As ActionResult
            If ModelState.IsValid Then
                _db.Warehouses.Add(warehouse)
                _db.SaveChanges()
                Return RedirectToAction("Index")
            End If
            Return View(warehouse)
        End Function

        ' GET: Warehouses/Edit/5
        Function Edit(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim warehouse As Warehouse = _db.Warehouses.Find(id)
            If IsNothing(warehouse) Then
                Return HttpNotFound()
            End If
            Return View(warehouse)
        End Function

        ' POST: Warehouses/Edit/5
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Edit(<Bind(Include:="ID,Code,Description,HasValue")> ByVal warehouse As Warehouse) As ActionResult
            If ModelState.IsValid Then
                _db.Entry(warehouse).State = EntityState.Modified
                _db.SaveChanges()
                Return RedirectToAction("Index")
            End If
            Return View(warehouse)
        End Function

        ' GET: Warehouses/Delete/5
        Function Delete(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim warehouse As Warehouse = _db.Warehouses.Find(id)
            If IsNothing(warehouse) Then
                Return HttpNotFound()
            End If
            Return View(warehouse)
        End Function

        ' POST: Warehouses/Delete/5
        <HttpPost()>
        <ActionName("Delete")>
        <ValidateAntiForgeryToken()>
        Function DeleteConfirmed(ByVal id As Integer) As ActionResult
            Dim warehouse As Warehouse = _db.Warehouses.Find(id)
            _db.Warehouses.Remove(warehouse)
            _db.SaveChanges()
            Return RedirectToAction("Index")
        End Function

        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If (disposing) Then
                _db.Dispose()
            End If
            MyBase.Dispose(disposing)
        End Sub
    End Class
End Namespace

File: Heat.ConvertedToC#/Controllers/WarehousesController.cs
using System.Linq;
using System.Web.Mvc;
using System.Net;
using Heat.Models;

namespace Heat.Controllers
{
    public class WarehousesController : System.Web.Mvc.Controller
	{


		private IHeatDBContext _db;
		public WarehousesController(IHeatDBContext context)
		{
			_db = context;
		}


		// GET: Warehouses
		public ActionResult Index()
		{
			return View(_db.Warehouses.ToList());
		}

		// GET: Warehouses/Details/5
		public ActionResult Details(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			Warehouse warehouse = _db.Warehouses.Find(id);
			if ((warehouse == null)) {
				return HttpNotFound();
			}
			return View(warehouse);
		}

		// GET: Warehouses/Create
		public ActionResult Create()
		{
			return View();
		}

		// POST: Warehouses/Create
		//To protect from overposting attacks, please enable the specific properties you want to bind to, for 
		//more details see http://go.microsoft.com/fwlink/?LinkId=317598.
		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Create(		[Bind(Include = "ID,Code,Description,HasValue")]
Warehouse warehouse)
		{
			if (ModelState.IsValid) {
				_db.Warehouses.Add(warehouse);
				_db.SaveChanges();
				return RedirectToAction("Index");
			}
			return View(warehouse);
		}

		// GET: Warehouses/Edit/5
		public ActionResult Edit(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			Warehouse warehouse = _db.Warehouses.Find(id);
			if ((warehouse == null)) {
				return HttpNotFound();
			}
			return View(warehouse);
		}

		// POST: Warehouses/Edit/5
		//To protect from overposting attacks, please enable the specific properties you want to bind to, for 
		//more details see http://go.microsoft.com/fwlink/?LinkId=317598.
		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Edit(		[Bind(Include = "ID,Code,Description,HasValue")]
Warehouse warehouse)
		{
			if (ModelState.IsValid) {
                _db.SetModified(warehouse);
				_db.SaveChanges();
				return RedirectToAction("Index");
			}
			return View(warehouse);
		}

		// GET: Warehouses/Delete/5
		public ActionResult Delete(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			Warehouse warehouse = _db.Warehouses.Find(id);
			if ((warehouse == null)) {
				return HttpNotFound();
			}
			return View(warehouse);
		}

		// POST: Warehouses/Delete/5
		[HttpPost()]
		[ActionName("Delete")]
		[ValidateAntiForgeryToken()]
		public ActionResult DeleteConfirmed(int id)
		{
			Warehouse warehouse = _db.Warehouses.Find(id);
			_db.Warehouses.Remove(warehouse);
			_db.SaveChanges();
			return RedirectToAction("Index");
		}

		protected override void Dispose(bool disposing)
		{
			if ((disposing)) {
				_db.Dispose();
			}
			base.Dispose(disposing);
		}
	}
}
